packages/atomic/src/components/common/generated-answer/generated-content/generated-markdown-content.spec.ts

```diff
@@ -0,0 +1,118 @@
+import DOMPurify from 'dompurify';
+import {html} from 'lit';
+import {afterEach, describe, expect, it, vi} from 'vitest';
+import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
+import {
+  type GeneratedMarkdownContentProps,
+  renderGeneratedMarkdownContent,
+} from './generated-markdown-content';
+import * as markdownUtils from './markdown-utils';
+
+vi.mock('dompurify');
+
+vi.mock('./markdown-utils', async (importOriginal) => {
+  const actual = await importOriginal<typeof markdownUtils>();
+  return {
+    ...actual,
+    transformMarkdownToHtml: vi.fn(actual.transformMarkdownToHtml),
+  };
+});
+
+describe('#renderGeneratedMarkdownContent', () => {
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  const locators = (element: Element) => ({
+    get generatedText() {
+      return element.querySelector('div[part="generated-text"]');
+    },
+  });
+
+  const renderComponent = async (
+    props: Partial<GeneratedMarkdownContentProps> = {}
+  ) => {
+    return await renderFunctionFixture(
+      html`${renderGeneratedMarkdownContent({
+        props: {
+          answer: 'Test answer',
+          isStreaming: false,
+          ...props,
+        },
+      })}`
+    );
+  };
+
+  it('should render a div element for the generated text in the document', async () => {
+    const element = await renderComponent();
+    const generatedText = locators(element).generatedText;
+    expect(generatedText).toBeInTheDocument();
+  });
+
+  it('should render with correct part attribute', async () => {
+    const element = await renderComponent();
+    const generatedText = locators(element).generatedText;
+
+    expect(generatedText).toHaveAttribute('part', 'generated-text');
+  });
+
+  it('should add "cursor" class when isStreaming is true', async () => {
+    const element = await renderComponent({
+      isStreaming: true,
+    });
+    const generatedText = locators(element).generatedText;
+
+    expect(generatedText).toHaveClass('cursor');
+  });
+
+  it('should not have "cursor" class when isStreaming is false', async () => {
+    const element = await renderComponent({
+      isStreaming: false,
+    });
+    const generatedText = locators(element).generatedText;
+
+    expect(generatedText).not.toHaveClass('cursor');
+  });
+
+  it('should call transformMarkdownToHtml with the answer', async () => {
+    const answer = '**bold** markdown';
+    await renderComponent({answer});
+
+    expect(markdownUtils.transformMarkdownToHtml).toHaveBeenCalledWith(answer);
+  });
+
+  it('should call transformMarkdownToHtml with empty string when answer is undefined', async () => {
+    await renderComponent({answer: undefined});
+
+    expect(markdownUtils.transformMarkdownToHtml).toHaveBeenCalledWith('');
+  });
```

Comment:
mocking of markdown-utils [4/4]
Use vi.mocked to get the mock, and for clarity, put it in a const at the top of your test.
Note, this is nitpiky at this stage whether you use a one-liner: expect(vi.mocked(funcName)).toHaveBeenCalledWith(arg)`

or

```ts
const mockedFuncName = vi.mocked(funcName);
expect(funcName).toHaveBeenCalledWith(arg)
```

As long as we are only talking about a single occurrence.

```suggestion
  it('should call transformMarkdownToHtml with the answer', async () => {
    const answer = '**bold** markdown';
    const mockedTransformMarkdownToHtml = vi.mocked(transformMarkdownToHtml);
    await renderComponent({answer});
    
    expect(mockedTransformMarkdownToHtml).toHaveBeenCalledWith(answer);
  });

  it('should call transformMarkdownToHtml with empty string when answer is undefined', async () => {
    const mockedTransformMarkdownToHtml = vi.mocked(transformMarkdownToHtml);
    await renderComponent({answer: undefined});

    expect(mockedTransformMarkdownToHtml).toHaveBeenCalledWith('');
  });
```